src/com/oltpbenchmark/api/Worker.java

```diff
@@ -321,6 +336,8 @@ public final void run() {
                         latencies.addLatency(type.getId(), start, end, this.id
                                 , phase.id);
                         intervalRequests.incrementAndGet();
+                        long latent = ((end - start + 500) / 1000);
+                        intervalLatencies.addAndGet(latent);
```

Comment:
This accumulates roundoff error. I'd just avoid dividing by 1000 until doing the average.